unstable/lens/hash-slice.rkt

```diff
@@ -0,0 +1,26 @@
+#lang racket/base
+
+(provide hash-slice-lens)
+
+(require lens/base/main)
+(module+ test
+  (require rackunit))
+
+(define (hash-slice-lens . ks)
+  (define (get hsh)
+    (for/hash ([k (in-list ks)])
+      (values k (hash-ref hsh k))))
+  (define (set hsh hsh2)
+    (define hsh1
+      (for/fold ([hsh hsh]) ([k (in-list ks)])
+        (hash-remove hsh k)))
+    (for/fold ([hsh hsh1]) ([(k v) (in-hash hsh2)])
+      (hash-set hsh k v)))
+  (make-lens get set))
+
+(module+ test
+  (check-equal? (lens-view (hash-slice-lens 'a 'c) (hash 'a 1 'b 2 'c 3))
+                (hash 'a 1 'c 3))
+  (check-equal? (lens-set (hash-slice-lens 'a 'c) (hash 'a 1 'b 2 'c 3) (hash 'd 4))
```

Comment:
This definition for setting makes hash lenses improper. It would make more sense if the hash lens only let you set the values of the keys it viewed. One of the core ideas of lenses is that you can only use them to manipulate their view, not anything else about their targets.